### PR TITLE
Onboard to validation framework

### PR DESCRIPTION
**Pull request by atalman**
Onboarding TorchData to Validation framework

### CODE REVIEW

File: .github/workflows/validate-binaries.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      os:
+        description: "Operating system to generate for (linux, windows, macos)"
+        required: true
+        type: choice
+        default: all
+        options:
+          - windows
+          - linux
+          - macos
+          - all
```
Comment (ejguan) [inline .github/workflows/validate-binaries.yml:40]: noob question: Is mac arm64 included?

**Review by ejguan — APPROVED** [review]
Stamp. Thank you for driving this effort to validate the compiled torchdata releases.

File: .github/scripts/validate_binaries.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+if [[ ${MATRIX_PACKAGE_TYPE} = "conda" ]]; then
+    conda install -y torchdata -c ${PYTORCH_CONDA_CHANNEL}
+else
+    pip install ${PYTORCH_PIP_PREFIX} torchdata --extra-index-url ${PYTORCH_PIP_DOWNLOAD_URL}
+fi
+
+python  ./test/smoke_tests/smoke_tests.py
```
Comment (ejguan) [inline .github/scripts/validate_binaries.sh:8]:
```suggestion
python  ./test/smoke_test/smoke_test.py
```

**Review by NivekT — APPROVED** [review]
@atalman Thanks! Feel free to import and land this. We can also do that for you if you'd like.

File: .github/scripts/validate_binaries.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
+if [[ ${MATRIX_PACKAGE_TYPE} = "conda" ]]; then
+    conda install -y torchdata -c ${PYTORCH_CONDA_CHANNEL}
+else
+    pip install ${PYTORCH_PIP_PREFIX} torchdata --extra-index-url ${PYTORCH_PIP_DOWNLOAD_URL}
+fi
+
+python  ./test/smoke_test/smoke_test.py
```
Comment (ejguan) [inline .github/scripts/validate_binaries.sh:8]: Can we do `python ./test/smoke_test/smoke_test.py --no-s3` when ${MATRIX_PACKAGE_TYPE} is `conda` on Windows?

Comment (ejguan) [reply .github/scripts/validate_binaries.sh]: Resolved as I just pushed the change to your branch.

**Pull request closed without merging**